core/domain/topic_domain.py

```diff
@@ -800,6 +806,11 @@ def validate(self, strict=False):
                 'Topic description should be at most %d characters, '
                 'received %s.' % (description_limit, self.description))
 
+        if not isinstance(self.category, python_utils.BASESTRING):
+            raise utils.ValidationError(
+                'Expected category to be a string, received %s'
+                % self.category)
```

Comment:
Is it supposed to be one of a predefined set of strings? If so, that needs to be validated, too.

Reply:
Done.